pkg/resolver/search.go

```diff
@@ -186,39 +191,50 @@ func (s *SearchResult) resolveItems() ([]map[string]interface{}, error) {
 
 func (s *SearchResult) getRelations() []SearchRelatedResult {
 	klog.Infof("Resolving relationships for [%d] uids.\n", len(s.uids))
+	var whereDs []exp.Expression
 
 	if len(s.input.RelatedKinds) > 0 {
-		// TODO: Use the RelatedKinds filter in the SQL query.
-		klog.Warning("TODO: The relationships query must use the provided kind filters.")
+		relatedKinds := pointerToStringArray(s.input.RelatedKinds)
+		whereDs = append(whereDs, goqu.C("destkind").In(relatedKinds).Expression())
 	}
+	//The level can be parameterized later, if needed, for applications
+	whereDs = append(whereDs, goqu.C("level").Eq(1)) // Add filter to select only level 1 relationships
 
 	//defining variables
 	items := []map[string]interface{}{}
 	var kindSlice []string
 	var kindList []string
 	var countList []int
 
-	relQuery := strings.TrimSpace(`WITH RECURSIVE
```

Comment:
Should we keep a comment with the original query for documentation?

Reply:
Done